src/main/java/org/codeforamerica/messaging/models/TemplateVariant.java

```diff
@@ -59,12 +70,53 @@ public String getTemplateName() {
         return template.getName();
     }
 
-    public String build(Function<TemplateVariant, String> templateFieldGetter, Map<String, String> templateParams) throws IOException {
+    public String build(Function<TemplateVariant, String> templateFieldGetter, Map<String, String> templateParams) {
+        com.github.jknack.handlebars.Template handlebarsTemplate = convertStringToTemplate(templateFieldGetter.apply(this));
+        if (handlebarsTemplate == null) {
+            return null;
+        }
+        Set<String> missingParams = getPlaceholdersInTemplateVariantField(handlebarsTemplate).stream()
+                .filter(not(templateParams::containsKey))
+                .collect(Collectors.toSet());
+        if (!missingParams.isEmpty()) {
+            throw new ParamsMissingException(missingParams);
+        }
+        try {
+            return handlebarsTemplate.apply(templateParams);
+        } catch (IOException e) {
+            throw new ParamsInvalidException(templateParams.keySet(), e);
+        }
+    }
+
+    com.github.jknack.handlebars.Template convertStringToTemplate(String templateString) {
```

Comment:
Would it be better if we called this `convertStringToHandlebarsTemplate`? The 2 meanings of `Template` were confusing to me.